### PR TITLE
feat: implement ability to add Fusion-only npm packages

### PR DESCRIPTION
**Pull request by Lodin**
Introduces an ability to re-add npm dependencies on `package.json` deletion for Fusion projects.

Fixes vaadin/fusion#52

### CODE REVIEW

**Comment by Lodin** [comment]
Due to a massive refactoring #11401 should be merged first.

**Comment by Lodin** [comment]
According to @alvarezguille, `lumo-css-framework` shouldn't be added to Fusion dependencies. Is it correct?

File: flow-server/src/main/java/com/vaadin/flow/server/frontend/NodeUpdater.java [hunk 1]
```diff
@@ -369,7 +369,7 @@ static void addVaadinDefaultsToJson(JsonObject json) {
 
         // Constructable style sheets is only implemented for chrome,
         // polyfill needed for FireFox et.al. at the moment
-        defaults.put("construct-style-sheets-polyfill", "2.4.16");
+        defaults.put("construct-style-sheets-polyfill", "3.0.0-0");
```
Comment (knoobie) [inline flow-server/src/main/java/com/vaadin/flow/server/frontend/NodeUpdater.java:372]: Was this change intended? Looks like something that should be a separate change for easier backporting.

**Comment by manolo** [comment]
Validation still failing, can you fix problems?

**Review by platosha — APPROVED** [review]
